Add/Basic VE subgraph

kafitz opened the pull request:
Resolves #37 
Resolves #45 

This PR adds a base subgraph for VE with:
- deposits
- withdrawals
- supply
- gage vote weights

Not sure if you're able to see it, but it's been deployed to thegraph Studio here:
1. https://thegraph.com/studio/subgraph/dfx-ve-test/
2. https://api.studio.thegraph.com/query/41366/dfx-ve-test/v0.0.12


Note: Sorry, it appears my lint-on-save is a bit zealous with converting quotes to double-quotes and adding semicolons.

chanharryhk commented:
> @chanharryhk It appears either the versions in this file should be downgraded, or the specVersion/apiVersion in the other subgraphs should be updated to use the latest `@graphprotocol` library.
> 
> If we update everything, then the CI should be updated to support at least node@14. Curious which path you'd prefer to go.

Yes good call I think we should update the CI its been broken for a while. Opened up an issue here https://github.com/dfx-finance/subgraph/issues/45

kafitz commented:
I added the following to fix the CI:
- bumped CI node version to 14
- bumped node dependency packages versions
- updated graph yaml's api and spec version*

*The build passes with the new spec and api versions, and I don't anticipate anything will break because of it. Just something to be aware of. The latest graph packages required API version of >= 0.0.5.

chanharryhk reviewed (approved):
LGTM
